### PR TITLE
Diagnostic API

### PR DESCRIPTION
**Pull request by ikitiki**
Diagnostic REST API provides profiling data via pprof, cluster and worker logs, cluster and controller statuses and configuration.

### CODE REVIEW

**Comment by ikitiki** [comment]
👍

**Comment by ikitiki** [comment]
👍

File: pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+}
+
+func (s *Server) clusters(w http.ResponseWriter, req *http.Request) {
+	var (
+		resp interface{}
+		err  error
+	)
+
+	if matches := clusterStatusURL.FindAllStringSubmatch(req.URL.Path, -1); matches != nil {
+		resp, err = s.controller.ClusterStatus(matches[0][1], matches[0][2])
```
Comment (alexeyklyukin) [inline pkg/apiserver/apiserver.go]: the numerical indexes are really hard to read/error prone. Perhaps wrap  SubexpNames in a function to extract matches by names?

**Comment by alexeyklyukin** [comment]
👍

**Comment by ikitiki** [comment]
👍

**Pull request merged**